Source/NSubstitute/Core/CallSpecificationFactory.cs

```diff
@@ -15,7 +15,7 @@ public ICallSpecification CreateFrom(ICall call, MatchArgs matchArgs)
         {
             var methodInfo = call.GetMethodInfo();
             var argumentSpecs = call.GetArgumentSpecifications();
-            var arguments = call.GetArguments();
+            var arguments = call.GetOriginalArguments();
```

Comment:
@dtchepak Sure, check [this](https://github.com/Zvirja/NSubstitute/blob/ea566ae65422ef768f02a7b2d5f6215d6b31a39e/Source/NSubstitute.Acceptance.Specs/CustomHandlersSpecs.cs#L92) and [this](https://github.com/Zvirja/NSubstitute/blob/ea566ae65422ef768f02a7b2d5f6215d6b31a39e/Source/NSubstitute.Acceptance.Specs/CustomHandlersSpecs.cs#L124) ones.